src/ds/d400/d400-options.h

```diff
@@ -57,7 +57,11 @@ namespace librealsense
     class auto_exposure_limit_option : public option_base
     {
     public:
-        auto_exposure_limit_option(hw_monitor& hwm, sensor_base* depth_ep, option_range range, std::shared_ptr<limits_option> exposure_limit_enable);
+        auto_exposure_limit_option( hw_monitor & hwm,
+                                    sensor_base * depth_ep,
```

Comment:
@OhadMeir do we want a weak_ptr here?

Reply:
I think we can remove this parameter completely, it is not used.
In other places we used a weak_ptr to check the sensor before sending HMC, but looking now I see that it is [checked by `locked_transfer`](https://github.com/IntelRealSense/librealsense/blob/development/src/hw-monitor.h#L219) so no need here.

We still need to do some order in all this, `command_transfer_over_xu` might need a weak_ptr and there are still options using direct reference to the sensor. In another PR.